recipes/lora_dpo_single_device.py

```diff
@@ -243,7 +243,7 @@ def setup(self, cfg: DictConfig) -> None:
         # Learning rate scheduler can only be set up after number of steps
         # has been computed
         self._lr_scheduler = self._setup_lr_scheduler(
-            cfg_lr_scheduler=cfg.lr_scheduler,
+            cfg_lr_scheduler=cfg.get("lr_scheduler", None),
```

Comment:
may be less indirection to just check directly if lr_scheduler exists and set to None here, instead of calling the setup method only to return None:
```
cfg_lr_scheduler = cfg.get("lr_scheduler", None)
self._lr_scheduler = self._setup_lr_scheduler(...) if cfg_lr_scheduler is not None else None
```

Reply:
That makes sense to me, Rafi, but upon second thought, i like the idea of handling everything inside of the setup_lr_scheduler, including the log_info. What do you think?

Reply:
Sure, no strong opinions